src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,348 @@
+import duke.command.ExitCommand;
+import duke.command.ListCommand;
+import duke.command.FindCommand;
+
+/**
+ * The `Parser` class represents a tool to make sense of user input.
+ * It provides methods to verify and understand user input, then create respective commands.
+ */
+public class Parser {
+
+    public enum TaskType {
+        BYE, LIST, MARK, UNMARK, TODO, DEADLINE, EVENT, DELETE, FIND, UNKNOWN
+    }
+
+    /**
+     * Returns a command after parsing user input.
+     * If the user input does not match any command, or any syntax error exists, throw exception.
+     *
+     * @param fullCommand User input.
+     * @return A command object to be executed.
+     * @throws IllegalArgumentException If the user input does not match any command, or any syntax error exists.
+     */
+    public static Command parse(String fullCommand) throws DukeException {
+        String[] splitedTask = fullCommand.split(" ");
+        Parser.TaskType taskType = Parser.TaskType.UNKNOWN;
+
+        // Empty command handler
+        try {
+            taskType = Parser.TaskType.valueOf(splitedTask[0].toUpperCase());
+        } catch (ArrayIndexOutOfBoundsException e) {
+            throw new DukeException("Sorry, we are not sync enough to communicate through empty command.");
+        } catch (IllegalArgumentException e) {
+            throw new DukeException("Syntax error, unknown command.");
+        }
+
+        switch (taskType) {
+        case BYE: {
+            return handleTaskBye(splitedTask);
+        }
+        case LIST: {
+            return handleTaskList(splitedTask);
+        }
+        case MARK: {
+            return handleTaskMark(splitedTask);
+        }
+        case UNMARK: {
+            return handleTaskUnmark(splitedTask);
+        }
+        case TODO: {
+            return handleTaskTodo(fullCommand, splitedTask);
+        }
+        case DEADLINE: {
+            return handleTaskDeadline(fullCommand, splitedTask);
+        }
+        case EVENT: {
+            return handleTaskEvent(fullCommand, splitedTask);
+        }
+        case DELETE: {
+            return handleTaskDelete(splitedTask);
+        }
+        case FIND: {
+            return handleTaskFind(fullCommand, splitedTask);
+        }
+        default: {
+            throw new DukeException("Syntax error, unknown command.");
+        }
+        }
+    }
+
+    /**
+     * Returns a `ExitCommand` after parsing user input.
+     * If any syntax error exists, throw exception.
+     *
+     * @param splitedTask Splited texts of the user input.
+     * @return A command object to be executed.
+     * @throws IllegalArgumentException If any syntax error exists.
+     */
+    public static Command handleTaskBye(String[] splitedTask) throws DukeException {
+        // Incorrect command syntax handler
+        if (splitedTask.length > 1) {
+            throw new DukeException("Syntax of bye: bye");
+        }
+        return new ExitCommand();
+    }
```

Comment:
Although you're following naming convention, perhaps you could make the function more intuitive. E.g. handleTaskExit since "bye" could potentially be replaced with any other word, but the action would remain the same "exit application"